compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -424,9 +424,12 @@ pub fn intern_const_alloc_recursive<
                 // something that cannot be promoted, which in constants means values that have
                 // drop glue, such as the example above.
                 InternKind::Constant => {
-                    ecx.tcx.sess.emit_err(UnsupportedUntypedPointer { span: ecx.tcx.span });
-                    // For better errors later, mark the allocation as immutable.
-                    alloc.mutability = Mutability::Not;
+                    // The Location does have leftover allocations, but that's fine, as we control it.
+                    if alloc_kind != MemoryKind::CallerLocation {
```

Comment:
Uh, this is bad. That code is already terrible and my attempts at cleaning it up in https://github.com/rust-lang/rust/pull/116745 are kind of stuck. I really don't like adding more complications here.